src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -5945,115 +5945,53 @@ class Program
-
-            Assert.NotEqual(0, exitCode);
-            Assert.Equal(0, tempFilePaths.Count);
-
-            tempFilePaths.Free();
-        }
-
-        [Fact]
-        public void IOFailure_OpenPdbTemp()
-        {
-            string sourcePath = MakeTrivialExe();
-            ArrayBuilder<string> tempFilePaths = ArrayBuilder<string>.GetInstance();
-            MockCSharpCompiler csc = MakeTrackingCsc(tempFilePaths, "/nologo", "/preferreduilang:en", "/debug", sourcePath);
-            string tempFilePath = null;
-
-            csc.FileOpen = (path, mode, access, share) =>
+            string exePath = Path.Combine(Path.GetDirectoryName(sourcePath), "test.exe");
+            var csc = new MockCSharpCompiler(null, _baseDirectory, new[] { "/nologo", $"/out:{exePath}", sourcePath });
+            csc.FileOpen = (file, mode, access, share) =>
             {
-                if (tempFilePaths.Count == 1)
+                if (file == exePath)
                 {
-                    tempFilePath = path;
-                    throw new IOException(path);
-                }
-                else
-                {
-                    return File.Open(path, mode, access, share);
+                    throw new IOException();
                 }
-            };
 
-            var outWriter = new StringWriter(CultureInfo.InvariantCulture);
-            int exitCode = csc.Run(outWriter);
-
-            var expectedOutput = string.Format("error CS1619: Cannot create temporary file -- {0}", tempFilePath);
-            Assert.Equal(expectedOutput, outWriter.ToString().Trim());
-
-            Assert.NotEqual(0, exitCode);
-            Assert.Equal(1, tempFilePaths.Count);
-
-            tempFilePaths.Free();
-        }
-
-        [Fact]
-        public void IOFailure_MoveAssemblyTemp()
-        {
-            string sourcePath = MakeTrivialExe();
-            ArrayBuilder<string> tempFilePaths = ArrayBuilder<string>.GetInstance();
-            MockCSharpCompiler csc = MakeTrackingCsc(tempFilePaths, "/nologo", "/preferreduilang:en", sourcePath);
-            csc.FileMove = (source, dest) =>
-            {
-                throw new IOException();
+                return File.Open(file, mode, access, share);
             };
 
             var outWriter = new StringWriter(CultureInfo.InvariantCulture);
-            int exitCode = csc.Run(outWriter);
-
-            var expectedOutput = string.Format("error CS2012: Cannot open '{0}' for writing -- 'I/O error occurred.'", Path.ChangeExtension(sourcePath, ".exe"));
-            Assert.Equal(expectedOutput, outWriter.ToString().Trim());
-
-            Assert.NotEqual(0, exitCode);
-            Assert.Equal(1, tempFilePaths.Count);
+            Assert.Equal(1, csc.Run(outWriter));
+            Assert.Contains($"error CS2012: Cannot open '{exePath}' for writing", outWriter.ToString());
 
-            tempFilePaths.Free();
+            System.IO.File.Delete(sourcePath);
+            System.IO.File.Delete(exePath);
```

Comment:
Just following the existing pattern in this file.  I agree it's weird but it's the pattern in use throughout.